packages/datastore-storage-adapter/src/ExpoSQLiteAdapter/ExpoSQLiteDatabase.ts

```diff
@@ -0,0 +1,285 @@
+					},
+					(_, error) => {
+						reject(error);
+						logger.warn(error);
+						return true;
+					}
+				);
+			});
+		});
+	}
+
+	public async save(
+		statement: string,
+		params: (string | number)[]
+	): Promise<void> {
+		return new Promise((resolve, reject) => {
+			this.db.transaction(tx => {
+				tx.executeSql(
+					statement,
+					params,
+					() => resolve(null),
+					(_, error) => {
+						reject(error);
+						logger.warn(error);
+						return true;
+					}
+				);
+			});
+		});
+	}
+
+	public async batchQuery<T = any>(
+		queryStatements: Set<ParameterizedStatement>
+	): Promise<T[]> {
+		return new Promise((resolveTx, rejectTx) => {
+			this.db.transaction(async tx => {
+				try {
+					const results = [];
+					// await to push all statement result into the results array.
+					await Promise.all(
+						[...queryStatements].map(
+							([statement, params]) =>
+								new Promise((resolve, reject) => {
+									tx.executeSql(
+										statement,
+										params,
+										(_, res) => {
+											results.push(res.rows._array[0]);
+											resolve(null);
+										},
+										(_, error) => {
+											reject(error);
+											logger.warn(error);
+											return true;
+										}
+									);
+								})
+						)
+					);
+					resolveTx(results);
+				} catch (error) {
+					rejectTx(error);
+				}
+			});
+		});
+	}
+
+	public async batchSave(
+		saveStatements: Set<ParameterizedStatement>,
+		deleteStatements?: Set<ParameterizedStatement>
+	): Promise<void> {
+		return new Promise((resolveTx, rejectTx) => {
+			try {
+				this.db.transaction(async tx => {
+					if (saveStatements) {
+						// await for all sql statments promises to resolve
+						await Promise.all(
+							[...saveStatements].map(
+								([statement, params]) =>
+									new Promise((resolve, reject) =>
+										tx.executeSql(
+											statement,
+											params,
+											(_, result) => {
+												resolve(null);
+											},
+											(_, error) => {
+												reject(error);
+												logger.warn(error);
+												return true;
+											}
+										)
+									)
+							)
+						);
+					}
+				});
+				if (deleteStatements) {
+					this.db.transaction(async tx => {
+						await Promise.all(
+							[...deleteStatements].map(
+								([statement, params]) =>
+									new Promise((resolve, reject) =>
+										tx.executeSql(
+											statement,
+											params,
+											(_, result) => {
+												resolve(null);
+											},
+											(_, error) => {
+												reject(error);
+												logger.warn(error);
+												return true;
+											}
+										)
+									)
+							)
+						);
+					});
+				}
+				resolveTx(null);
+			} catch (error) {
+				rejectTx(error);
+			}
+		});
+	}
+
+	public async selectAndDelete<T = any>(
+		queryParameterizedStatement: ParameterizedStatement,
+		deleteParameterizedStatement: ParameterizedStatement
```

Comment:
Nit: Imo, it would be better to align these with `saveStatements` or `deleteStatements`:
```suggestion
		queryStatement: ParameterizedStatement,
		deleteStatement: ParameterizedStatement
```
Same goes for the `selectAndDelete` in the `SQLiteDatabase.ts`

Reply:
`queryStatement` and `deleteStatement` would interfere and be confusing when destructuring `queryParameterizedStatement` and `deleteParameterizedStatement` in the later step. eg.

```
		const [queryStatement, queryParams] = queryParameterizedStatement;
		const [deleteStatement, deleteParams] = deleteParameterizedStatement;

```

Reply:
I think it is much clearer for future reader if we use `deleteParameterizedStatements` instead of just `deleteStatements` because it makes it clearer that the each individual item inside `deleteParameterizedStatements` is of type `ParameterizedStatement`. However, your concern regarding lengthening the names of the contents of `Set` is valid. I think we can rename the arguments of `batchSave` to use `saveParameterizedStatements` and `deleteParameterizedStatements`. However, as we are directly de-structuring the items inside `saveParameterizedStatements` and `deleteParameterizedStatements`, we do not need to declare `const deleteParameterizedStatement`. So, that it looks like below (simplified for demonstration).

```
this.db.transaction(async tx => {
	[...(saveParameterizedStatements ?? [])].map(([saveStatement, params]) => {
		tx.executeSql(saveStatement, params);
	});
});

```

Reply:
Okay, i think you are aligning with me? What I am getting is that in the current form, there no consistent naming pattern

Reply:
Agreed Caleb, there were some inconsistent naming of the function arguments. I'll rename the function arguments to follow more consistent naming pattern. Thanks for pointing this out.